CONTRIBUTING.md

```diff
@@ -24,19 +24,79 @@ Please follow these steps when making pull requests (PRs):
     Please give the PR a descriptive name.
 
 1.  As you make progress on your PR, keep your branch up-to-date with the `master` branch which
-        may have been updated after starting your PR.
+        may have been updated *after* starting your PR.
     Your PR **MUST** be updated to reflect changes to `master` in order to be merged.
-    Use the following procedure for updating your branch:
+    Use the following procedure for updating your branch and are ready to commit your changes:
 
     <!--- language: lang-shell -->
 
-        # While on your PR branch <issue-branch>
+        ## While on your PR branch <issue-branch> hosted at <your-remote> repository:
         git commit
         git checkout master
-        git pull origin --rebase
+        git pull <illixr-remote> --rebase                           ## Line A
```

Comment:
Why pull into master instead of pulling directly into your feature branch?

Pulling into master should be done just before starting a new feature branch, which might entail adding a new section here called "starting a new feature branch" or something.

Reply:
> As you make progress on your PR, keep your branch up-to-date with the `master` branch which may have been updated *after* starting your PR.

The idea here is that pulling and rebasing into`master` should be done frequently, and for all ongoing work on existing and new branches. Using this scheme, all updates to `master` are fast-forwards, and `master` is always up-to-date. The process for continuing work on a branch and starting work on a new branch off of `master` become the same, except for the case of true rebase conflicts from bringing an existing feature branch in line with `master`. I think this becomes more important as there are more concurrent feature branches in progress; the state of one's local repository can be assumed to be clean and up-to-date as branches are context switched.